apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java

```diff
@@ -245,4 +256,10 @@ int getErrorCount() {
         T withConnection(HttpURLConnection connection) throws IOException;
     }
 
+    public synchronized void overrideApmServerUrls(List<URL> serverUrls) {
```

Comment:
Just inline this to the `onChange` method. Doesn't have to be synchronized.

Reply:
fixed